src/main/java/mat/server/MeasureLibraryServiceImpl.java

```diff
@@ -1764,6 +1764,13 @@ public final SaveMeasureResult saveOrUpdateMeasure(ManageMeasureDetailModel mode
                 measureSet.setId(UUID.randomUUID().toString());
                 measurePackageService.save(measureSet);
             }
+
+            if (pkg.getMeasureDetails() == null) {
+                MeasureDetails measureDetails = new MeasureDetails();
```

Comment:
Is there a way to cover that logic with unit test?

Reply:
I spent some time yesterday trying to write unit test for this, but given the current state of the code, it is exceedingly hard to write a good test.

Reply:
ok, it was not a blocking comment.